src/idx_html.htl

```diff
@@ -0,0 +1 @@
+${content.json.string}
```

Comment:
I think we should use the JSON pipeline for this. Instead of requesting `README.idx.html`, you'd request `README.idx.json`. The `htl` file can be dropped and you'd transform the `idx_html.pre.js` into a `idx_json.js` file. You can use this as a starting point: https://github.com/adobe/helix-cli/blob/master/test/integration/src/json.js

Reply:
i agree that the request for something with a `.json` extension to correspond with the `Content-Type: application/json` is the right path to go. However, I think we currently have a general issue where we make a choice of which pipeline to use based on the extension of the request path.
I think given that the `idx.json` represents a view of the DOM or more specifically of the HTML output to be indexed, this should use the html pipeline instead of the json pipeline.

Furthermore i cannot really come up with a usecase anymore that would require a separate json pipeline alltogether, especially considering that most JSON usecases that i can think of are better served with the GraphQL interface to begin with. I currently think that the most obvious `.json` usecase would be the `.idx.json` which clearly should operate on the HTML pipeline, which makes me think that we might not need multiple pipelines anymore.

Reply:
When looking at the pipelines, it makes sense to unify the front part of the pipelines, so that JSON, HTML and XML behave more alike, but the back part (output) formatting should probably still stay different, just to ensure things like mine types, OpenWhisk encoding expectations, and so on.